### PR TITLE
3.0

### PR DESCRIPTION
**Pull request by mickhansen**

### CODE REVIEW

**Comment by mikrofusion** [comment]
@mickhansen I'd love to get https://github.com/sequelize/sequelize/pull/3705 included in the 3.0 release too.  Without it sequelize seems to have issues adding indexes to tables when using schemas (using postgres).  This PR fixes the last of the bugs I've found so far having to do with creating indexes on a table with a schema.
If not, no worries, but thought I'd propose this while I saw this branch out there.

**Comment by mickhansen** [comment]
@mikrofusion Does it break any kind of BC? If not i dont mind spinning out a quick 3.0.1 release after i merge yours.

This won't be ready till tommorow evening likely, so we can probably get your fix in too. But this release is a priority because of a security concern.

**Comment by mikrofusion** [comment]
@mickhansen sounds good.  No break in any kind of BC with my changes.  
I'm good either way on my end, just wanted to make you aware since the PR is ready and should fix indexing on tables with schemas.
Thanks!

File: lib/sequelize.js [hunk 1]
```diff
@@ -667,22 +632,16 @@ module.exports = (function() {
    *
    * @see {Model#build} for more information about callee.
    */
-  Sequelize.prototype.query = function(sql, callee, options, replacements) {
+  Sequelize.prototype.query = function(sql, callee, options) {
```
Comment (BridgeAR) [inline lib/sequelize.js]: Don't you want to deprecate/remove the callee too? It's already possible to use options.instance instead (undocumented though).

File: lib/sequelize.js [hunk 2]
```diff
@@ -667,22 +632,16 @@ module.exports = (function() {
...
-      }
-    } else if (arguments.length === 4) {
-      deprecated('passing raw query replacements as the 4th argument to sequelize.query is deprecated. Please use options.replacements instead');
-      options.replacements = replacements;
-    } else if (arguments.length !== 3) {
-      options = { raw: true };
+
+    if (_.isPlainObject(callee)) {
+      options = callee;
+      callee = options.instance || options.model || undefined;
```
Comment (BridgeAR) [inline lib/sequelize.js]: The undefined at the end is not necessary (I know I wrote that originally :D).

File: lib/model.js [hunk 3]
```diff
@@ -748,51 +747,67 @@ module.exports = (function() {
...
-    // For sanity findOne will never return if no options are passed
+  Model.prototype.findById = function(param, options) {
+    // For sanity findById will never return if no options are passed
     if ([null, undefined].indexOf(param) !== -1) {
       return Promise.resolve(null);
     }
 
-    var options = {};
+    options = options || {};
+    delete options.where;
```
Comment (BridgeAR) [inline lib/model.js]: Using `options.where = undefined;` would be nicer (if nothing checks for the options keys - but _.defaults does not).

File: test/integration/include.test.js [hunk 4]
```diff
@@ -190,7 +190,7 @@ describe(Support.getTestDialectTeaser('Include'), function() {
         , Users;
 
       Groups = User.belongsToMany(Group, { through: 'UserGroup' });
-      Users = Group.belongsToMany(User), { through: 'UserGroup' };
```
Comment (BridgeAR) [inline test/integration/include.test.js]: How did this pass earlier? :smile:

File: changelog.md [hunk 5]
```diff
@@ -1,5 +1,28 @@
-# Next
+# 3.0.0
+
+3.0.0 cleans up a lot of deprecated code, making it easier for us to develop and maintain features in the future.
```
Comment (BridgeAR) [inline changelog.md]: Maybe add a comment that sequelize is going to use semver versions in future?

Comment (mickhansen) [inline changelog.md]: That will go in the readme.

File: lib/sequelize.js (hunk 1 above)
Comment (mickhansen) [inline lib/sequelize.js]: Not sure, already buried in a shit tonne of work getting what i've removed so far to work, 5 hours so far and not green.

File: lib/sequelize.js (hunk 2 above)
Comment (mickhansen) [inline lib/sequelize.js]: Probably not but it does add some sort of clarity

File: lib/model.js (hunk 3 above)
Comment (mickhansen) [inline lib/model.js]: I think `delete` is more descriptive actually.

File: test/integration/include.test.js (hunk 4 above)
Comment (mickhansen) [inline test/integration/include.test.js]: No clue ;p

File: lib/sequelize.js (hunk 1 above)
Comment (BridgeAR) [inline lib/sequelize.js]: I'll help you if you want me too. I think it's really good to get rid of all the old stuff (besides the original issue that required some of these changes).

Comment (mickhansen) [inline lib/sequelize.js]: Yep, lets get rid of it.

Comment (mickhansen) [inline lib/sequelize.js]: Will also make it easier to distinguish between `options.instance` and `options.model` as callee.

**Pull request closed without merging**